### ISSUE TITLE
Update Coverage for Sun's Java Style

### ISSUE THREAD

**Issue by Aman-Baliyan** [checkstyle/checkstyle#20811]
Style Guide:[Code Conventions for the Java™ Programming Language](https://www.oracle.com/java/technologies/javase/codeconventions-contents.html)([cached page](https://checkstyle.sourceforge.io/styleguides/sun-code-conventions-19990420/CodeConvTOC.doc.html))
Coverage Page: [src/site/xdoc/sun_style.xml](https://checkstyle.org/sun_style.html)
Config File: [src/main/resources/sun_checks.xml](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/sun_checks.xml)

### Context
Sun's Java Style Checkstyle Coverage was last updated on 2019 after that time new checks are added but this report is not updated.
We need to update [src/site/xdoc/sun_style.xml](https://checkstyle.org/sun_style.html) and its embedded config [src/main/resources/sun_checks.xml](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/sun_checks.xml)
 We need to add checks for each rule in the style guide and complete all ??? entries in the coverage page - similar to how it is done for [Google's Java Style Checkstyle Coverage](https://checkstyle.org/google_style.html).

All sections and rules are already listed in the [coverage page](https://checkstyle.org/sun_style.html). Rows marked with ??? need to be completed.

### How to contribute
1. Pick a subsection from the tracking list below, comment "I am working on  `X.Y - Subsection Name`" on this issue, and create a **child issue** referencing this parent issue.

   * **Title format:** `Add checks for Sun Style [X.Y - Subsection Name]`
   * In the child issue, list all the rules and corresponding checks that will enforce that rule (see [example child issue](https://github.com/checkstyle/checkstyle/issues/20812)).
   *  Add a checklist in the child issue to track progress per rule.
2. Wait for the issue to get **approved** before starting work to avoid unnecessary iterations on PRs.
3. Make **one PR per check/rule**. If a subsection has multiple rules, do NOT put them all in one PR. Submit a separate PR for each check so it is easy to review and merge.
4. In each PR clearly mention:
   * Which rule it covers (e.g., "Files longer than 2000 lines are cumbersome and should be avoided.")
   * Which check is being added
   * Reference the child issue
5. Each PR should include:
   * The check added to `src/main/resources/sun_checks.xml` with proper configuration
   * The corresponding ??? row updated in `src/site/xdoc/sun_style.xml`, following the format used in [google_style.xml](https://checkstyle.org/google_style.html)
   * Integration tests following the folder structure below
6. Update your child issue checklist as PRs are raised.

**Note**: If a check needed to cover a section is already present in sun_checks.xml then only update coverage page and add test suite for that rule.

Also after adding the check remove check from this list - https://github.com/checkstyle/checkstyle/blob/777653dd93ce1a9906841601b84530dd9e3072ea/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L180

### Integration test structure
Tests must follow the same section-based folder structure used by Google style tests. The existing structure is:

```
src/it/resources/com/sun/checkstyle/test/chapter{N}{description}/rule{NNN}{description}/Input*.java
src/it/java/com/sun/checkstyle/test/chapter{N}{description}/rule{NNN}{description}/*Test.java
```
Use [`src/it/resources/com/google/checkstyle/test/`](https://github.com/checkstyle/checkstyle/tree/master/src/it/resources/com/google/checkstyle/test) as the reference for how test input files and test classes are organized by chapter and rule.

**Note**: For the integration tests, we will not follow modulewise testing. We will follow chapterwise testing like Google style, one test class per section, not per check.  Please follow the Google style test structure as reference.

### Reference
Use [`google_style.xml`](https://checkstyle.org/google_style.html) and [`google_checks.xml`](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml) as the formatting and structure reference.

### Subsection tracking

**3. File Organizations**

| Subsection | Child Issue | Status |
| :--- | :--- | :--- |
| 3. File Organizations | | | |
| 3.1 Java Source Files | | | |
| 31.1. Beginning Comments | | | |
| 3.1.2 Package And Import Names | | | |
| 3.1.3 Class And Interface Declarations | | | |

**4. Indentation**

| Subsection | Child Issue | Status |
| :--- | :--- | :--- |
| 4. Indentation | | | |
| 4.1 Line Length | | | |
| 4.2 Wrapping Lines | | | |

**5. Comments**

| Subsection | Child Issue | Status |
| :--- | :--- | :--- |
| 5. Comments | | | |
| 5.1 Implementation Comment Formats | | | |
| 5.1.1 Block Comments | | | |
| 5.1.2 Single-Line Comments | | | |
| 5.1.3 Trailing Comments | | | |
| 5.1.4 End-Of-Line Comments | | | |
| 5.2 Documentation Comments | | | |

**6. Declarations**

| Subsection | Child Issue | Status |
| :--- | :--- | :--- |
| 6. Declarations | | | |
| 6.1 Number Per Line | | | |
| 6.2 Initialization | | | |
| 6.3 Placement | | | |
| 6.4 Class and Interface Declarations | | | |

**7. Statements**

| Subsection | Child Issue | Status |
| :--- | :--- | :--- |
| 7. Statements | | | |
| 7.1 Simple Statements | | | |
| 7.2 Compound Statements | | | |
| 7.3 return Statements | | | |
| 7.4 if, if-else, if else-if else Statements | | | |
| 7.5 for Statements | | | |
| 7.6 while Statements | | | |
| 7.7 do-while Statements | | | |
| 7.8 switch Statements | | | |
| 7.9 try-catch Statements | | | |

**8. White Space**

| Subsection | Child Issue | Status |
| :--- | :--- | :--- |
| 8. White Space | | | |
| 8.1 Blank Lines | | | |
| 8.2 Blank Spaces | | | |

**9. Naming Conventions**

| Subsection | Child Issue | Status |
| :--- | :--- | :--- |
| 9. Naming Conventions | | | |

**10. Programming Practices**

| Subsection | Child Issue | Status |
| :--- | :--- | :--- |
| 10. Programming Practices | | | |
| 10.1 Providing Access to Instance and Class Variables | | | |
| 10.2 Referring to Class Variables and Methods | | | |
| 10.3 Constants | | | |
| 10.4 Variable Assignments | | | |
| 10.5 Miscellaneous Practices | | | |
| 10.5.1 Parentheses | | | |
| 10.5.2 Returning Values | | | |
| 10.5.3 Expressions before `?' in the Conditional Operator | | | |
| 10.5.4 Special Comments | | | |

**Comment by DakshRJain737** [comment]
I am working on 3.0 File Organizations

**Comment by Aman-Baliyan** [comment]
> I am working on 2.1 File Suffixes

@DakshRJain737 please choose another section because there is no specific rule under chapter 2. You can try ->  3.0 -  File Organizations

**Comment by hemantch01** [comment]
Hi @romani, I would like to work on 4.1 Line Length . Is there anything else I should check, or any extra context before I open the child issue?

**Comment by Aman-Baliyan** [comment]
> Hi [@romani](https://github.com/romani), I would like to work on 4.1 Line Length . Is there anything else I should check, or any extra context before I open the child issue?

@hemantch01  you can see https://github.com/checkstyle/checkstyle/pull/20921 for refrence nothing else required.

**Comment by hemantch01** [comment]
hey @Aman-Baliyan i should wait for this issue to get approved?

**Comment by Aman-Baliyan** [comment]
> hey [@Aman-Baliyan](https://github.com/Aman-Baliyan) i should wait for this issue to get approved?

you can submit a pr issue is good.

**Comment by MaybeSomeone-arc18** [comment]
I am working on 4.2 Wrapping Lines. I will submit a separate PR for the SeparatorWrap rule, following child issue #21528.

**Comment by hemantch01** [comment]
hi @MaybeSomeone-arc18 i already started working on this can you pick other issues which are open, from issues?? plz

**Comment by MaybeSomeone-arc18** [comment]
Thanks for letting me know. I wasn't aware that 4.2 was already being worked on, so I'll step back from this subsection and pick another available issue.
